### PR TITLE
Improve Cody logging agent protocol

### PR DESCRIPTION
**Pull request by pkukielka**
## Changes

Fixes logging capability in agent.
Previously messages logged to output channel were visible only if added using `append` or `appendLine`.
Now `trace` / `debug` / `info` / `warn` / `error` functions are also logged properly.

## Test plan

As in https://github.com/sourcegraph/jetbrains/pull/2583

### CODE REVIEW

File: agent/src/vscode-shim.ts [hunk 1]
```diff
@@ -530,32 +530,27 @@ export namespace UriString {
 }
 
 function outputChannel(name: string): vscode.LogOutputChannel {
+    function notifyAgent(level: DebugMessageLogLevel, message: string, ...args: any[]): void {
+        if (agent) {
```
Comment (mkondratek) [inline agent/src/vscode-shim.ts:534]: There is some contradiction in this method. It is called notifyAgent but we seem to check if we use agent inside of it.

maybe `s/notifyAgent/notifyAgentIfApplicable/g`?

no hard feelings about it tho. feel free to leave it as it is if it makes sense to you

**Review by mkondratek — APPROVED** [review]
don't we need to regenerate the recordings?

**Comment by pkukielka** [comment]
> don't we need to regenerate the recordings?

We need, and they are regenerated :)

File: agent/src/vscode-shim.ts (hunk 1 above)
Comment (pkukielka) [reply agent/src/vscode-shim.ts:534]: This check does not verify if we are in the 'agent' mode, it just check if agent object is already initialized.
`vscode-shim.ts` is used only for agent.

Comment (mkondratek) [reply agent/src/vscode-shim.ts:534]: oh, makes sense.

**Pull request merged**